src/Microsoft.Data.SqlClient/tests/UnitTests/PoolTests.cs

```diff
@@ -0,0 +1,539 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if NET8_0_OR_GREATER
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient.NetCore.UnitTests.Util;
+using Microsoft.Data.SqlClientX;
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.NetCore.UnitTests.Handlers.Prelogin
+{
+    public class PoolTests
+    {
+        private TestBase testBase;
+
+        public PoolTests()
+        {
+            testBase = new TestBase();
+        }
+
+
+        [Fact]
+        public async Task MinPoolSize_equals_MaxPoolSize()
+        {
+            await using var dataSource = testBase.CreateDataSource(csb =>
+            {
+                csb.MinPoolSize = 30;
+                csb.MaxPoolSize = 30;
+            });
+            await using var conn = await dataSource.OpenConnectionAsync();
+        }
+
+        [Fact]
+        public void MinPoolSize_bigger_than_MaxPoolSize_throws()
```

Comment:
Also usually naming the tests like `MethodName_OptionalCondition_Outcome` is a bit more readable
Eg, for this one I'd rename it `MinPoolSize_BiggerThanMaxPoolSize_Throws`